logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -171,8 +167,8 @@ private String compactJson(final String json) throws IOException {
     }
 
     // this wouldn't catch spaces in json, but that's ok for our use case here
-    private boolean isAlreadyCompacted(final String json) {
-        return PRETTY_PRINT.matchesNoneOf(json);
+    final static boolean isAlreadyCompacted(final String json) {
+        return !json.matches("(?s).*[\n\r\t].*");
```

Comment:
Pattern could be static final field, and I think the pattern can be simplified to just `[\n\r\t]`.